tests/test_native_ops.py

```diff
@@ -188,3 +188,54 @@ def test_apply_amplitude_damp_single_wire(
         simulator_device_1_wire.apply([ops.AmplitudeDamp(*par, wires=[0])])
 
         assert np.allclose(simulator_device_1_wire.state, expected_density_matrix, **tol)
+
+    @pytest.mark.parametrize("input",
+        [
+            [1, 0, 0, 0],
+            [0, 1, 0, 0],
+            [0, 0, 1, 0],
+            [0, 0, 0, 1],
+        ]
+    )
+    def test_apply_iswap(self, tol, input, shots, analytic):
+        """Tests that applying the CPhase gate yields the expected output."""
+        device = SimulatorDevice(2, shots=shots, analytic=analytic)
+
+        iswap_mat = np.array([[1, 0, 0, 0],
+                              [0, 0, 1j, 0],
+                              [0, 1j, 0, 0],
+                              [0, 0, 0, 1]])
+
+        expected = input @ iswap_mat
```

Comment:
Why is the gate being applied to the RHS? Is this only working because the input is the identity matrix (and hence commutes with all gates)?

Reply:
:facepalm: